releasenotes/notes/add-fit-model-and-remove-curve_fitter-arg-4a0effb5f9b88ba9.yaml

```diff
@@ -0,0 +1,24 @@
+---
+features:
+  - |
+    :attr:`CurveAnalysis.fit_model` and :attr:`CurveAnalysis.parameters` have been
+    added to the curve analysis and its subclasses. These properties return
+    :class:`FitModel` instance that represents a fit function of the analysis
+    and parameters to be fit with the analysis, respectively.
+  - |
+    New classes :class:`SingleFitFunction` and :class:`CompositeFitFunction` have been
+    added to the curve analysis module. These classes abstract the curve definition
+    hard-coded in the class attribute :attr:`CurveAnalysis.__series__` as a
+    Python function-like object. This may help user to grasp the fitting model
+    without reading the code.
+upgrade:
+  - |
+    Analysis option `curve_fitter` of the :class:`CurveAnalysis` has be removed
+    because of the serialization problem. To use custom curve fitting library,
+    one must directly override the class method :meth:`CurveAnalysis.curve_fit`.
+developer:
+  - |
+    :meth:`CurveAnalysis.curve_fit` has been added to the curve analysis and
+    its subclasses. Now you can directly access to the core fitting code
+    with bare numpy arrays representing data to be fit.
+    This may help debugging of new fit function.
```

Comment:
```suggestion
    This may help debugging of new fit functions.
```

Reply:
Please add a reference to an example that demonstrates how to use the new feature for debugging fit functions.

Reply:
There is no example, but user can directly input ndarrays without constructing `ExperimentData` container. Since there is no user guide how to construct experiment data, it is very tough to test fit functions in the curve analysis without reading the code of experiment data which is super long. I can remove
> This may help debugging of new fit functions.

If I need to write mode docs just for it.